xtask/src/commands/rextendr_document.rs

```diff
@@ -0,0 +1,41 @@
+//! This invokes `rextendr::document()` within `tests/extendrtests`.
+//!
+//! It uses the vendored `rextendr` in the repository as the source package.
+//!
+//! 1. Ensure that `git submodule update --init` was invoked once, as to setup
+//! the vendored `rextendr` package.
+//! 2. `devtools` must be installed on system.
+//!
+//!
+//! The idea here is to be able to develop `rextendr` alongside `extendr`,
+//! as well as ease the development of extendr.
+//!
+use std::error::Error;
+
+use crate::extendrtests::with_absolute_path::{swap_extendr_api_path, R_FOLDER_PATH};
+use xshell::{cmd, Shell};
+
+pub(crate) fn run(shell: &Shell) -> Result<(), Box<dyn Error>> {
+    let _document_handle = swap_extendr_api_path(shell)?;
+
+    run_rextendr_document(shell)
+}
+
+fn run_rextendr_document(shell: &Shell) -> Result<(), Box<dyn Error>> {
+    let _r_path = shell.push_dir(R_FOLDER_PATH);
+
+    //FIXME: check if `../../rextendr` is available, and report back
+    // if it is not, then it is due to lack of `git submodule update --init`
+    // which should be
+    //FIXME: test if `devtools` is available, and report back if not
+
+    cmd!(shell, "Rscript")
+        .args([
+            "-e",
+            r#"devtools::load_all("../../rextendr")"#,
```

Comment:
This is highly opinionated. Why not check for installed `{rextendr}` and use it?

Reply:
yes! I've been experimenting with adding `libR-sys` and `rextendr` as git submodules to the repo. This has worked wonderfully, because then if I need to change something either crate, I can do that, and I can hinge that change on a certain branch/commit/fork. Then a PR to extendr can be made, when it works with the published version of either crate.

Also, this is supposed to be a helper to a contributor that are working on all packages extendr. I hope we can find a good way to include this.

Reply:
Ok, makes sense. I never used submodules so this is something new to me. Perhaps better documentation & checks will help?

Reply:
Done